ladybug/_datacollectionbase.py

```diff
@@ -940,6 +940,18 @@ def is_mutable(self):
         """Boolean denoting whether the data collection is mutable."""
         return self._mutable
 
+    def __key(self):
+        return (
+            self.header, self.values, self.datetimes, self.validated_a_period,
+            self._enumeration
```

Comment:
You really don't want to check that the `_enumeration` is equal in order to tell if two collections are equal.  This `_enumeration` is only there to help with the conversion between immutable and mutable collections and you can see that it doesn't even appear in `__slots__`. It's just some data that needs to be auto-generated after loading the class but only in situations where you actually need to convert between mutable and immutable collections.

Reply:
Also, I would recommend overwriting this `__key` method for the `HourlyContinuousDataCollection`. Since the majority of data collections we use are of that type and there's no need to check that the `datetimes` and `validated_a_period `are equal on that class. This will generally be much faster for the common cases we use the data collection for.